File: Level 3/네트워크/solution.js
function solution(n, computers) {
  let answer = 0;
  const visited = Array(n).fill(false);

  visited.forEach((isVisited, computer) => {
    if (!isVisited) {
      dfs(computer, computers, visited);
      answer++;
    }
  });

  return answer;
}

function dfs(computer, computers, visited) {
  visited[computer] = true;

  computers[computer].forEach((isConnected, currentComputer) => {
    if (isConnected && !visited[currentComputer]) {
      dfs(currentComputer, computers, visited);
    }
  });
}

File: Level 3/네트워크/solution.py
def dfs(computer, computers, visited):
    visited[computer] = True
    for current_computer, is_connected in enumerate(computers[computer]):
        if not visited[current_computer] and is_connected:
            dfs(current_computer, computers, visited)


def solution(n, computers):
    answer = 0
    visited = [False] * n

    for computer in range(n):
        if not visited[computer]:
            dfs(computer, computers, visited)
            answer += 1

    return answer
